src/eth/storage/permanent_storage.rs

```diff
@@ -0,0 +1,109 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+
+use super::EthStorageError;
+use crate::eth::primitives::Account;
+use crate::eth::primitives::Address;
+use crate::eth::primitives::Block;
+use crate::eth::primitives::BlockNumber;
+use crate::eth::primitives::BlockSelection;
+use crate::eth::primitives::Execution;
+use crate::eth::primitives::ExecutionConflicts;
+use crate::eth::primitives::Hash;
+use crate::eth::primitives::LogFilter;
+use crate::eth::primitives::LogMined;
+use crate::eth::primitives::Slot;
+use crate::eth::primitives::SlotIndex;
+use crate::eth::primitives::StoragePointInTime;
+use crate::eth::primitives::TransactionMined;
+
+/// Permanent (committed) storage operations
+// TODO: add Metrified method
+#[async_trait]
+pub trait PermanentStorage: Send + Sync {
+    // Retrieves the last mined block number.
+    async fn read_current_block_number(&self) -> anyhow::Result<BlockNumber>;
+
+    /// Atomically increments the block number, returning the new value.
+    async fn increment_block_number(&self) -> anyhow::Result<BlockNumber>;
```

Comment:
Probably we don't want to increment the block number directly in the permanent storage.

The block number will be updated in the permanent storage when the temporary storage is commited.